datacube/api/core.py

```diff
@@ -447,12 +447,14 @@ def empty_func(measurement_):
         for name, coord in coords.items():
             result[name] = coord
         for name, coord in geobox.coordinates.items():
-            result[name] = (name, coord.values, {'units': coord.units, 'resolution': coord.resolution})
+            result[name] = (name, coord.values, {'units': coord.units,
+                                                 'resolution': coord.resolution,
+                                                 'crs': str(geobox.crs)})
 
         for measurement in measurements:
             data = data_func(measurement)
             attrs = measurement.dataarray_attrs()
-            attrs['crs'] = geobox.crs
+            attrs['crs'] = str(geobox.crs)
```

Comment:
for performance and safety reasons I would convert geobox.crs to string once, that way you also save on memory as you won;t be storing multiple identical but different strings.